File: Study/Programming Skills/05. Repeated Substring Pattern/Go/main.go
package main 

import "fmt"

func main() {
  fmt.Println(repeatedSubstringPattern("abab")) // true
  fmt.Println(repeatedSubstringPattern("aba")) // false
  fmt.Println(repeatedSubstringPattern("abcabcabcabc")) // true
}

func repeatedSubstringPattern(s string) bool {
  n := len(s)
  for i := 1; i <= n/2; i++ {
    if n%i == 0 {
      if s[:i] == s[i:2*i] {
        if s == repeat(s[:i], n/i) {
          return true
        }
      }
    }
  }
  return false
}

File: Study/Programming Skills/05. Repeated Substring Pattern/JavaScript/main.js
/**
 * @param {string} s
 * @return {boolean}
 */
var repeatedSubstringPattern = function (s) {
  try {
    return (s + s).slice(1, -1).includes(s);
  } catch (error) {
    console.log('error :>> ', error);
  }
};

// Test Cases
console.log(repeatedSubstringPattern('abab') === true);
console.log(repeatedSubstringPattern('aba') === false);
console.log(repeatedSubstringPattern('abcabcabcabc') === true);
